src/praisonai/praisonai/agents_generator.py

```
@@ -444,11 +445,12 @@ def _init_observability(self, framework: str) -> None:
         Args:
             framework: The framework name for tagging
         """
+        # Initialize AgentOps if configured (extracted from old _select_autogen_version)
         agentops_api_key = os.getenv("AGENTOPS_API_KEY")
         if agentops_api_key:
             try:
                 import agentops
-                agentops.init(agentops_api_key, default_tags=[framework])
+                agentops.init(agentops_api_key, default_tags=[framework_name])
```

Ruff (0.15.17)

[error] Undefined name `framework_name`

(F821)